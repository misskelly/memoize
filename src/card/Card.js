import React, { Component } from 'react';
import flipArrow from '../Styles/flipArrow.png';
import trash from '../Styles/trash.png';
import Button from './button/Button'

export default class Card extends Component {
  constructor(props) {
    super(props);
    
    this.state = {
      // currentCard: null,
      termFaceUp: true,
    }
  }


  render() {

    return (
      
      <section className='featuredCard' id={this.props.cardId}>
        <article className='termSide card'>
          <h4 className='termText'>{this.props.term}</h4>
          <div className='iconBtnWrapper' >
            <img src={trash} alt='I am a little trash can! Click me to remove this card from the deck.' role='button' className='icon trash'/>
            <img src={flipArrow} alt='I am a curved arrow! Click me to filp the card over and see the answer.' role='button' className='icon arrow'/>
          </div>
        </article>
        <article className='definitionSide card'>
          <p className='definitionText'>{this.props.definition}</p>
          <form className='buttons'>
            <Button type='removeBtn'
                    buttonText='Nailed it!'
                    lable='Remove this flashcard, this one is memorized!'
                    removeFromDeck={this.props.removeFromDeck}
                    cardId={this.props.cardId} 
                    getRandomCard={this.props.getRandomCard}
                    />
            <Button type='almostBtn'
                    buttonText='So Close!'
                    lable='Better come back to that one later.' 
                    getRandomCard={this.props.getRandomCard}
                    />
            <Button type='wrongAnswerBtn'
                    buttonText='Not a clue.'
                    lable='Yikes, I need extra review on that one!' 
                    getRandomCard={this.props.getRandomCard}

                    // Better add a duplicate!'
                    // addDuplicate={props.addDuplicate} 
                    cardId={this.props.cardId} 
                    />
          </form>
        </article>
      </section>
    )

  }

}