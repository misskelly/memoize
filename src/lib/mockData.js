const mockCards = [
  {
    "categoryId": 1,
    "cardNumber": 101,
    "term": "React",
    "definition": "A client-side JavaScript framework that allows you to easily and efficiently manipulate the DOM based on application data and how it changes in response to user interaction"
  },
  {
    "categoryId": 1,
    "cardNumber": 102,
    "term": "The Virtual DOM",
    "definition": "An in-memory object that represents a DOM structure and can be manipulated with JavaScript before updating the real DOM"
  },
  {
    "categoryId": 1,
    "cardNumber": 103,
    "term": "JSX",
    "definition": "A mix of JavaScript and XML that facilitates rendering the appropriate HTML, lives inside a component"
  },
  {
    "categoryId": 2,
    "cardNumber": 205,
    "term": "Enzyme",
    "definition": "A JavaScript Testing utility for React that makes it easier to test your React Components' output.  It allows test files to be more readable and provides us with helper methods for testing common pieces of functionality."
  },
]

const mockCurrentCard = {
  "categoryId": 1,
  "cardNumber": 103,
  "term": ".simulate()",
  "definition": "Simulates an event on an element by passing the event name string in as an argument."
}


module.exports = {
  mockCards,
  mockCurrentCard
}