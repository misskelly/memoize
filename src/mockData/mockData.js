const mockCards = [
  {
    "categoryId": 1,
    "cardNumber": 101,
    "term": "React",
    "definition": "A client-side JavaScript framework that allows you to easily and efficiently manipulate the DOM based on application data and how it changes in response to user interaction"
  },
  {
    "categoryId": 1,
    "cardNumber": 102,
    "term": "The Virtual DOM",
    "definition": "An in-memory object that represents a DOM structure and can be manipulated with JavaScript before updating the real DOM"
  },
  {
    "categoryId": 1,
    "cardNumber": 103,
    "term": "JSX",
    "definition": "A mix of JavaScript and XML that facilitates rendering the appropriate HTML, lives inside a component"
  },
  {
    "categoryId": 1,
    "cardNumber": 104,
    "term": "Component",
    "definition": "Reusable pieces of code that represent templates for a particular instance of a UI element. They are either functions or an extended ES6 class"
  }
]

const mockCurrentCard = {
  "categoryId": 2,
  "cardNumber": 214,
  "term": ".simulate()",
  "definition": "Simulates an event on an element by passing the event name string in as an argument."
}


module.exports = {
  mockCards,
  mockCurrentCard
}