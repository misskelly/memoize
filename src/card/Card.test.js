import React from 'react';
import Card from './Card';
import { mockCurrentCard } from '../mockData/mockData'
import { shallow } from 'enzyme';


const mockRemoveFromDeck = jest.fn();
const mockGetRandomCard = jest.fn();

describe('Card', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Card cardId={mockCurrentCard.cardNumber}
            term={mockCurrentCard.term}
            definition={mockCurrentCard.definition}
            getRandomCard={mockGetRandomCard}
            removeFromDeck={mockRemoveFromDeck} />
            // addDuplicate={mockCurrentCard.addDuplicate} />
    )
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});