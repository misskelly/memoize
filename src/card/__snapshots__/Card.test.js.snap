// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    cardId={214}
    definition="Simulates an event on an element by passing the event name string in as an argument."
    getRandomCard={[MockFunction]}
    removeFromDeck={[MockFunction]}
    term=".simulate()"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <article
          className="termSide card"
        >
          <h4
            className="termText"
          >
            .simulate()
          </h4>
          <div
            className="iconBtnWrapper"
          >
            <img
              alt="I am a little trash can! Click me to remove this card from the deck."
              className="icon trash"
              role="button"
              src="trash.png"
            />
            <img
              alt="I am a curved arrow! Click me to filp the card over and see the answer."
              className="icon arrow"
              role="button"
              src="flipArrow.png"
            />
          </div>
        </article>,
        <article
          className="definitionSide card"
        >
          <p
            className="definitionText"
          >
            Simulates an event on an element by passing the event name string in as an argument.
          </p>
          <form
            className="buttons"
          >
            <Button
              buttonText="Nailed it!"
              cardId={214}
              getRandomCard={[MockFunction]}
              lable="Remove this flashcard, this one is memorized!"
              removeFromDeck={[MockFunction]}
              type="removeBtn"
            />
            <Button
              buttonText="So Close!"
              getRandomCard={[MockFunction]}
              lable="Better come back to that one later."
              type="almostBtn"
            />
            <Button
              buttonText="Not a clue."
              cardId={214}
              getRandomCard={[MockFunction]}
              lable="Yikes, I need extra review on that one!"
              type="wrongAnswerBtn"
            />
          </form>
        </article>,
      ],
      "className": "featuredCard",
      "id": 214,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4
              className="termText"
            >
              .simulate()
            </h4>,
            <div
              className="iconBtnWrapper"
            >
              <img
                alt="I am a little trash can! Click me to remove this card from the deck."
                className="icon trash"
                role="button"
                src="trash.png"
              />
              <img
                alt="I am a curved arrow! Click me to filp the card over and see the answer."
                className="icon arrow"
                role="button"
                src="flipArrow.png"
              />
            </div>,
          ],
          "className": "termSide card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": ".simulate()",
              "className": "termText",
            },
            "ref": null,
            "rendered": ".simulate()",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="I am a little trash can! Click me to remove this card from the deck."
                  className="icon trash"
                  role="button"
                  src="trash.png"
                />,
                <img
                  alt="I am a curved arrow! Click me to filp the card over and see the answer."
                  className="icon arrow"
                  role="button"
                  src="flipArrow.png"
                />,
              ],
              "className": "iconBtnWrapper",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "I am a little trash can! Click me to remove this card from the deck.",
                  "className": "icon trash",
                  "role": "button",
                  "src": "trash.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "I am a curved arrow! Click me to filp the card over and see the answer.",
                  "className": "icon arrow",
                  "role": "button",
                  "src": "flipArrow.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": "div",
          },
        ],
        "type": "article",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="definitionText"
            >
              Simulates an event on an element by passing the event name string in as an argument.
            </p>,
            <form
              className="buttons"
            >
              <Button
                buttonText="Nailed it!"
                cardId={214}
                getRandomCard={[MockFunction]}
                lable="Remove this flashcard, this one is memorized!"
                removeFromDeck={[MockFunction]}
                type="removeBtn"
              />
              <Button
                buttonText="So Close!"
                getRandomCard={[MockFunction]}
                lable="Better come back to that one later."
                type="almostBtn"
              />
              <Button
                buttonText="Not a clue."
                cardId={214}
                getRandomCard={[MockFunction]}
                lable="Yikes, I need extra review on that one!"
                type="wrongAnswerBtn"
              />
            </form>,
          ],
          "className": "definitionSide card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Simulates an event on an element by passing the event name string in as an argument.",
              "className": "definitionText",
            },
            "ref": null,
            "rendered": "Simulates an event on an element by passing the event name string in as an argument.",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Button
                  buttonText="Nailed it!"
                  cardId={214}
                  getRandomCard={[MockFunction]}
                  lable="Remove this flashcard, this one is memorized!"
                  removeFromDeck={[MockFunction]}
                  type="removeBtn"
                />,
                <Button
                  buttonText="So Close!"
                  getRandomCard={[MockFunction]}
                  lable="Better come back to that one later."
                  type="almostBtn"
                />,
                <Button
                  buttonText="Not a clue."
                  cardId={214}
                  getRandomCard={[MockFunction]}
                  lable="Yikes, I need extra review on that one!"
                  type="wrongAnswerBtn"
                />,
              ],
              "className": "buttons",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "buttonText": "Nailed it!",
                  "cardId": 214,
                  "getRandomCard": [MockFunction],
                  "lable": "Remove this flashcard, this one is memorized!",
                  "removeFromDeck": [MockFunction],
                  "type": "removeBtn",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "buttonText": "So Close!",
                  "getRandomCard": [MockFunction],
                  "lable": "Better come back to that one later.",
                  "type": "almostBtn",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "buttonText": "Not a clue.",
                  "cardId": 214,
                  "getRandomCard": [MockFunction],
                  "lable": "Yikes, I need extra review on that one!",
                  "type": "wrongAnswerBtn",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "form",
          },
        ],
        "type": "article",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <article
            className="termSide card"
          >
            <h4
              className="termText"
            >
              .simulate()
            </h4>
            <div
              className="iconBtnWrapper"
            >
              <img
                alt="I am a little trash can! Click me to remove this card from the deck."
                className="icon trash"
                role="button"
                src="trash.png"
              />
              <img
                alt="I am a curved arrow! Click me to filp the card over and see the answer."
                className="icon arrow"
                role="button"
                src="flipArrow.png"
              />
            </div>
          </article>,
          <article
            className="definitionSide card"
          >
            <p
              className="definitionText"
            >
              Simulates an event on an element by passing the event name string in as an argument.
            </p>
            <form
              className="buttons"
            >
              <Button
                buttonText="Nailed it!"
                cardId={214}
                getRandomCard={[MockFunction]}
                lable="Remove this flashcard, this one is memorized!"
                removeFromDeck={[MockFunction]}
                type="removeBtn"
              />
              <Button
                buttonText="So Close!"
                getRandomCard={[MockFunction]}
                lable="Better come back to that one later."
                type="almostBtn"
              />
              <Button
                buttonText="Not a clue."
                cardId={214}
                getRandomCard={[MockFunction]}
                lable="Yikes, I need extra review on that one!"
                type="wrongAnswerBtn"
              />
            </form>
          </article>,
        ],
        "className": "featuredCard",
        "id": 214,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4
                className="termText"
              >
                .simulate()
              </h4>,
              <div
                className="iconBtnWrapper"
              >
                <img
                  alt="I am a little trash can! Click me to remove this card from the deck."
                  className="icon trash"
                  role="button"
                  src="trash.png"
                />
                <img
                  alt="I am a curved arrow! Click me to filp the card over and see the answer."
                  className="icon arrow"
                  role="button"
                  src="flipArrow.png"
                />
              </div>,
            ],
            "className": "termSide card",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": ".simulate()",
                "className": "termText",
              },
              "ref": null,
              "rendered": ".simulate()",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    alt="I am a little trash can! Click me to remove this card from the deck."
                    className="icon trash"
                    role="button"
                    src="trash.png"
                  />,
                  <img
                    alt="I am a curved arrow! Click me to filp the card over and see the answer."
                    className="icon arrow"
                    role="button"
                    src="flipArrow.png"
                  />,
                ],
                "className": "iconBtnWrapper",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "I am a little trash can! Click me to remove this card from the deck.",
                    "className": "icon trash",
                    "role": "button",
                    "src": "trash.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "I am a curved arrow! Click me to filp the card over and see the answer.",
                    "className": "icon arrow",
                    "role": "button",
                    "src": "flipArrow.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
              ],
              "type": "div",
            },
          ],
          "type": "article",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="definitionText"
              >
                Simulates an event on an element by passing the event name string in as an argument.
              </p>,
              <form
                className="buttons"
              >
                <Button
                  buttonText="Nailed it!"
                  cardId={214}
                  getRandomCard={[MockFunction]}
                  lable="Remove this flashcard, this one is memorized!"
                  removeFromDeck={[MockFunction]}
                  type="removeBtn"
                />
                <Button
                  buttonText="So Close!"
                  getRandomCard={[MockFunction]}
                  lable="Better come back to that one later."
                  type="almostBtn"
                />
                <Button
                  buttonText="Not a clue."
                  cardId={214}
                  getRandomCard={[MockFunction]}
                  lable="Yikes, I need extra review on that one!"
                  type="wrongAnswerBtn"
                />
              </form>,
            ],
            "className": "definitionSide card",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Simulates an event on an element by passing the event name string in as an argument.",
                "className": "definitionText",
              },
              "ref": null,
              "rendered": "Simulates an event on an element by passing the event name string in as an argument.",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Button
                    buttonText="Nailed it!"
                    cardId={214}
                    getRandomCard={[MockFunction]}
                    lable="Remove this flashcard, this one is memorized!"
                    removeFromDeck={[MockFunction]}
                    type="removeBtn"
                  />,
                  <Button
                    buttonText="So Close!"
                    getRandomCard={[MockFunction]}
                    lable="Better come back to that one later."
                    type="almostBtn"
                  />,
                  <Button
                    buttonText="Not a clue."
                    cardId={214}
                    getRandomCard={[MockFunction]}
                    lable="Yikes, I need extra review on that one!"
                    type="wrongAnswerBtn"
                  />,
                ],
                "className": "buttons",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "buttonText": "Nailed it!",
                    "cardId": 214,
                    "getRandomCard": [MockFunction],
                    "lable": "Remove this flashcard, this one is memorized!",
                    "removeFromDeck": [MockFunction],
                    "type": "removeBtn",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "buttonText": "So Close!",
                    "getRandomCard": [MockFunction],
                    "lable": "Better come back to that one later.",
                    "type": "almostBtn",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "buttonText": "Not a clue.",
                    "cardId": 214,
                    "getRandomCard": [MockFunction],
                    "lable": "Yikes, I need extra review on that one!",
                    "type": "wrongAnswerBtn",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "form",
            },
          ],
          "type": "article",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
